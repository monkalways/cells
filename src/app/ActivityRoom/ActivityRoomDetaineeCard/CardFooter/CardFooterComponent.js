import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Button, CardActions, Typography, withStyles,
} from '@material-ui/core';
import { withRouter } from 'react-router-dom';
import { compose } from 'recompose';

import CheckInDetaineeDialog from './CheckInDetaineeDialog';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    id: PropTypes.string.isRequired,
    activityRoomName: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  isCheckingIn: PropTypes.bool.isRequired,
  isCheckingInSuccess: PropTypes.bool.isRequired,
  usage: PropTypes.string.isRequired,
  onCheckIn: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  checkingInDetaineeId: PropTypes.string,
};

const defaultProps = {
  checkingInDetaineeId: null,
};

const LOCATION_IN_TRANSIT = 'in transit';
const LOCATION_IN_PROGRESS = 'in progress';

export class CardFooterComponent extends Component {
  state = {
    isDialogOpen: false,
  };

  componentWillReceiveProps(nextProps) {
    this.setState({
      isDialogOpen:
        nextProps.isCheckingIn
        && !nextProps.isCheckingInSuccess
        && nextProps.detainee.id === nextProps.checkingInDetaineeId,
    });
  }

  handleInButtonClick = () => {
    this.setState({
      isDialogOpen: true,
    });
  };

  handleOutButtonClick = () => {
    const { history, detainee, usage } = this.props;
    const first = 'activity-rooms';
    history.push(`/detainees/${detainee.id}?first=${first}&second=${usage}`);
  };

  handleClose = () => {
    this.setState({
      isDialogOpen: false,
    });
  };

  handleCheckIn = () => {
    const { onCheckIn, detainee, usage } = this.props;
    onCheckIn(detainee.id, usage);
  };

  renderButton = () => {
    const { detainee, classes } = this.props;
    const { location } = detainee;
    if (location) {
      if (location.toLowerCase().includes(LOCATION_IN_TRANSIT)) {
        return (
          <Button
            id="inButton"
            variant="contained"
            size="large"
            className={classes.inButton}
            onClick={this.handleInButtonClick}
          >
            IN
          </Button>
        );
      }
      if (location.toLowerCase().includes(LOCATION_IN_PROGRESS)) {
        return (
          <Button
            id="outButton"
            variant="contained"
            size="large"
            className={classes.outButton}
            onClick={this.handleOutButtonClick}
          >
            OUT
          </Button>
        );
      }
    }
    return null;
  };

  render() {
    const {
      classes,
      isAuthenticated,
      detainee,
      isCheckingIn,
      usage,
    } = this.props;
    const { isDialogOpen } = this.state;
    return (
      <CardActions className={classes.actions}>
        {isAuthenticated && (
          <React.Fragment>
            <Typography variant="h6" className={classes.heading}>
              {detainee.activityRoomName}
            </Typography>
            {this.renderButton()}
            <CheckInDetaineeDialog
              isDialogOpen={isDialogOpen}
              detainee={detainee}
              isCheckingIn={isCheckingIn}
              usage={usage}
              onClose={this.handleClose}
              onCheckIn={this.handleCheckIn}
            />
          </React.Fragment>
        )}
      </CardActions>
    );
  }
}

CardFooterComponent.propTypes = propTypes;
CardFooterComponent.defaultProps = defaultProps;

export default compose(
  withStyles((theme) => ({
    actions: {
      display: 'flex',
      height: theme.spacing.unit * 9,
    },
    heading: {
      width: theme.spacing.unit * 8,
      marginRight: theme.spacing.unit * 4,
    },
    inButton: {
      backgroundColor: '#2895F2',
      color: '#FFFFFF',
      fontSize: theme.typography.h6.fontSize,
      '&:hover': {
        backgroundColor: '#0E47A1',
      },
    },
    outButton: {
      backgroundColor: '#558A3A',
      color: '#FFFFFF',
      fontSize: theme.typography.h6.fontSize,
      '&:hover': {
        backgroundColor: '#456336',
      },
    },
  })),
  withRouter,
)(CardFooterComponent);
