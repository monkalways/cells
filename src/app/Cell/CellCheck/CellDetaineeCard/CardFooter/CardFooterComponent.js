import React from 'react';
import PropTypes from 'prop-types';
import { CardActions, IconButton, withStyles } from '@material-ui/core';

import WellnessVisualIcon from '../../../../images/WellnessVisual.png';
import WellnessVerbalIcon from '../../../../images/WellnessVerbal.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellCheck: PropTypes.shape({
    visual: PropTypes.bool.isRequired,
    verbal: PropTypes.bool.isRequired,
  }),
  onVisualClick: PropTypes.func,
  onVerbalClick: PropTypes.func,
};

const defaultProps = {
  cellCheck: null,
  onVisualClick: null,
  onVerbalClick: null,
};

const CardFooterComponent = ({
  classes,
  cellCheck,
  onVisualClick,
  onVerbalClick,
}) => (
  <CardActions className={classes.actions} disableActionSpacing>
    {cellCheck && (
      <React.Fragment>
        <IconButton
          className={
            cellCheck.visual ? classes.leftButtonSelected : classes.leftButton
          }
          onClick={onVisualClick}
        >
          <img
            src={WellnessVisualIcon}
            alt="wellness visual"
            className={classes.img}
          />
        </IconButton>
        <IconButton
          className={
            cellCheck.verbal ? classes.rightButtonSelected : classes.rightButton
          }
          onClick={onVerbalClick}
        >
          <img
            src={WellnessVerbalIcon}
            alt="wellness verbal"
            className={classes.img}
          />
        </IconButton>
      </React.Fragment>
    )}
  </CardActions>
);

CardFooterComponent.propTypes = propTypes;
CardFooterComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  actions: {
    display: 'flex',
    height: theme.spacing.unit * 9,
  },
  leftButton: {
    margin: theme.spacing.unit,
    opacity: 0.2,
  },
  leftButtonSelected: {
    margin: theme.spacing.unit,
  },
  rightButton: {
    marginLeft: theme.spacing.unit * 5,
    margin: theme.spacing.unit,
    opacity: 0.2,
  },
  rightButtonSelected: {
    marginLeft: theme.spacing.unit * 5,
    margin: theme.spacing.unit,
  },
  img: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
}))(CardFooterComponent);
