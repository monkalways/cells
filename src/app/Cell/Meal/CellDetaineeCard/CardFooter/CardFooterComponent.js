import React from 'react';
import PropTypes from 'prop-types';
import { CardActions, IconButton, withStyles } from '@material-ui/core';

import MealAcceptIcon from '../../../../images/MealAccept.png';
import MealDeclineIcon from '../../../../images/MealDecline.png';
import NotApplicableIcon from '../../../../images/NotApplicable.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  meal: PropTypes.shape({
    accept: PropTypes.bool.isRequired,
    reject: PropTypes.bool.isRequired,
    notApplicable: PropTypes.bool.isRequired,
  }),
  onAcceptClick: PropTypes.func.isRequired,
  onRejectClick: PropTypes.func.isRequired,
  onNotApplicableClick: PropTypes.func.isRequired,
};

const defaultProps = {
  meal: null,
};

const CardFooterComponent = ({
  classes,
  meal,
  onAcceptClick,
  onRejectClick,
  onNotApplicableClick,
}) => (
  <CardActions className={classes.actions} disableActionSpacing>
    {meal && (
      <React.Fragment>
        <IconButton
          className={meal.accept ? classes.buttonSelected : classes.button}
          onClick={onAcceptClick}
        >
          <img src={MealAcceptIcon} alt="meal accept" className={classes.img} />
        </IconButton>
        <IconButton
          className={meal.reject ? classes.buttonSelected : classes.button}
          onClick={onRejectClick}
        >
          <img
            src={MealDeclineIcon}
            alt="meal decline"
            className={classes.img}
          />
        </IconButton>
        <IconButton
          className={
            meal.notApplicable ? classes.buttonSelected : classes.button
          }
          onClick={onNotApplicableClick}
        >
          <img
            src={NotApplicableIcon}
            alt="not applicable"
            className={classes.img}
            disabled
          />
        </IconButton>
      </React.Fragment>
    )}
  </CardActions>
);

CardFooterComponent.propTypes = propTypes;
CardFooterComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  actions: {
    display: 'flex',
    height: theme.spacing.unit * 9,
  },
  button: {
    margin: 0,
    opacity: 0.2,
  },
  buttonSelected: {
    margin: 0,
  },
  img: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
}))(CardFooterComponent);
