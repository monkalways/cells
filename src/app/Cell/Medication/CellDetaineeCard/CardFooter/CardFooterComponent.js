import React from 'react';
import PropTypes from 'prop-types';
import { CardActions, IconButton, withStyles } from '@material-ui/core';

import MedicineAcceptIcon from '../../../../images/MedicineAccept.png';
import MedicineDeclineIcon from '../../../../images/MedicineDecline.png';
import NotApplicableIcon from '../../../../images/NotApplicable.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({}).isRequired,
  medication: PropTypes.shape({
    accept: PropTypes.bool.isRequired,
    reject: PropTypes.bool.isRequired,
    notApplicable: PropTypes.bool.isRequired,
  }),
  onAcceptClick: PropTypes.func.isRequired,
  onRejectClick: PropTypes.func.isRequired,
  onNotApplicableClick: PropTypes.func.isRequired,
};

const defaultProps = {
  medication: null,
};

export const CardFooterComponent = ({
  classes,
  cellDetainee,
  medication,
  onAcceptClick,
  onRejectClick,
  onNotApplicableClick,
}) => (
  <CardActions className={classes.actions} disableActionSpacing>
    {!cellDetainee.location
      && cellDetainee.isUnderMedication
      && medication && (
        <React.Fragment>
          <IconButton
            id="acceptButton"
            className={
              medication.accept ? classes.buttonSelected : classes.button
            }
            onClick={onAcceptClick}
          >
            <img
              src={MedicineAcceptIcon}
              alt="Medicine accept"
              className={classes.img}
            />
          </IconButton>
          <IconButton
            id="rejectButton"
            className={
              medication.reject ? classes.buttonSelected : classes.button
            }
            onClick={onRejectClick}
          >
            <img
              src={MedicineDeclineIcon}
              alt="Medicine decline"
              className={classes.img}
            />
          </IconButton>
          <IconButton
            id="notApplicableButton"
            className={
              medication.notApplicable ? classes.buttonSelected : classes.button
            }
            onClick={onNotApplicableClick}
          >
            <img
              src={NotApplicableIcon}
              alt="not applicable"
              className={classes.img}
              disabled
            />
          </IconButton>
        </React.Fragment>
    )}
  </CardActions>
);

CardFooterComponent.propTypes = propTypes;
CardFooterComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  actions: {
    display: 'flex',
    height: theme.spacing.unit * 9,
  },
  button: {
    margin: 0,
    opacity: 0.2,
  },
  buttonSelected: {
    margin: 0,
  },
  img: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
}))(CardFooterComponent);
