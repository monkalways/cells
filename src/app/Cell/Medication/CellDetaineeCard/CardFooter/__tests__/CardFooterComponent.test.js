import React from 'react';
import { shallow } from 'enzyme';

import { CardFooterComponent } from '../CardFooterComponent';

describe('CardFooterComponent', () => {
  let onAcceptClick;
  let onRejectClick;
  let onNotApplicableClick;
  let medication;

  beforeEach(() => {
    onAcceptClick = jest.fn();
    onRejectClick = jest.fn();
    onNotApplicableClick = jest.fn();
    medication = null;
  });

  const setup = () => {
    const classes = {
      actions: 'actions',
      buttonSelected: 'buttonSelected',
      button: 'button',
      img: 'img',
    };
    return shallow(<CardFooterComponent
      classes={classes}
      onAcceptClick={onAcceptClick}
      onRejectClick={onRejectClick}
      onNotApplicableClick={onNotApplicableClick}
      medication={medication}
    />);
  };

  it('should not render any buttons if medication is null', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render accept, reject and not-applicable buttons if medication is not null', () => {
    medication = {
      accept: true,
      reject: false,
      notApplicable: false,
    };
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('#acceptButton')).toExist();
    expect(wrapper.find('#rejectButton')).toExist();
    expect(wrapper.find('#notApplicableButton')).toExist();
  });

  it('should render accept button as selected if accept medication is selected', () => {
    medication = {
      accept: true,
      reject: false,
      notApplicable: false,
    };
    const wrapper = setup();

    expect(wrapper.find('#acceptButton')).toHaveProp(
      'className',
      'buttonSelected',
    );
    expect(wrapper.find('#rejectButton')).toHaveProp('className', 'button');
    expect(wrapper.find('#notApplicableButton')).toHaveProp(
      'className',
      'button',
    );
  });

  it('should render reject button as selected if reject medication is selected', () => {
    medication = {
      accept: false,
      reject: true,
      notApplicable: false,
    };
    const wrapper = setup();

    expect(wrapper.find('#rejectButton')).toHaveProp(
      'className',
      'buttonSelected',
    );
    expect(wrapper.find('#acceptButton')).toHaveProp('className', 'button');
    expect(wrapper.find('#notApplicableButton')).toHaveProp(
      'className',
      'button',
    );
  });

  it('should render not-applicable button as selected if not-applicable is selected', () => {
    medication = {
      accept: false,
      reject: false,
      notApplicable: true,
    };
    const wrapper = setup();

    expect(wrapper.find('#notApplicableButton')).toHaveProp(
      'className',
      'buttonSelected',
    );
    expect(wrapper.find('#acceptButton')).toHaveProp('className', 'button');
    expect(wrapper.find('#rejectButton')).toHaveProp('className', 'button');
  });

  it('should handle accept click', () => {
    medication = {
      accept: true,
      reject: false,
      notApplicable: false,
    };
    const wrapper = setup();

    wrapper.find('#acceptButton').simulate('click');
    expect(onAcceptClick).toBeCalled();
  });

  it('should handle reject click', () => {
    medication = {
      accept: false,
      reject: true,
      notApplicable: false,
    };
    const wrapper = setup();

    wrapper.find('#rejectButton').simulate('click');
    expect(onRejectClick).toBeCalled();
  });

  it('should handle not-applicable click', () => {
    medication = {
      accept: false,
      reject: false,
      notApplicable: true,
    };
    const wrapper = setup();

    wrapper.find('#notApplicableButton').simulate('click');
    expect(onNotApplicableClick).toBeCalled();
  });
});
