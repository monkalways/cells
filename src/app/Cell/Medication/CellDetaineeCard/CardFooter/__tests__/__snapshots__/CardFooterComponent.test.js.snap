// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooterComponent should not render any buttons if medication is null 1`] = `
<WithStyles(CardActions)
  className="actions"
  disableActionSpacing={true}
/>
`;

exports[`CardFooterComponent should render accept, reject and not-applicable buttons if medication is not null 1`] = `
<WithStyles(CardActions)
  className="actions"
  disableActionSpacing={true}
>
  <WithStyles(IconButton)
    className="buttonSelected"
    id="acceptButton"
    onClick={[MockFunction]}
  >
    <img
      alt="Medicine accept"
      className="img"
      src="MedicineAccept.png"
    />
  </WithStyles(IconButton)>
  <WithStyles(IconButton)
    className="button"
    id="rejectButton"
    onClick={[MockFunction]}
  >
    <img
      alt="Medicine decline"
      className="img"
      src="MedicineDecline.png"
    />
  </WithStyles(IconButton)>
  <WithStyles(IconButton)
    className="button"
    id="notApplicableButton"
    onClick={[MockFunction]}
  >
    <img
      alt="not applicable"
      className="img"
      disabled={true}
      src="NotApplicable.png"
    />
  </WithStyles(IconButton)>
</WithStyles(CardActions)>
`;
